server/src/main/java/org/opensearch/cluster/SnapshotsInProgress.java

```diff
@@ -937,7 +937,8 @@ public enum State {
         STARTED((byte) 1, false),
         SUCCESS((byte) 2, true),
         FAILED((byte) 3, true),
-        ABORTED((byte) 4, false);
+        ABORTED((byte) 4, false),
+        PARTIAL((byte) 5, false);
```

Comment:
@aggarwalShivani we changing Public API here and I believe it will cause issues in mixed clusters:  the older versions will not be able to desearialize the `State` enumeration since the new member `PARTIAL` is not known to them

Reply:
@reta  Thanks for your feedback. I'm making this change, please let me know if this aligns well - 

```
                       case PARTIAL:
                            if (Version.CURRENT.onOrAfter(Version.V_3_0_0))
                                state = SnapshotsInProgress.State.PARTIAL;
                            else
                                // Setting SUCCESS in older versions for backward compatibility
                                state = SnapshotsInProgress.State.SUCCESS;
```

Reply:
@aggarwalShivani apologies, the change is not right, it should be done in `Entry::writeTo(StreamOutput)` and `Entry::Entry(StreamInput)` , please let me know if you need help

Reply:
@reta Thanks for the feedback, apologies for not getting it earlier. 
I'm afraid I will need some help here :(

Is this how the change should be made?
```
        public void writeTo(StreamOutput out) throws IOException {
            snapshot.writeTo(out);
            ....
            if ((out.getVersion().before(Version.V_3_0_0)) && state.value == (byte) 5){
                    out.writeByte((byte) 2);
            } else {
                    out.writeByte(state.value());
            }
```

Also, I didnt really understand the change to be made in the private Entry(StreamInput in) method.

Reply:
@aggarwalShivani  yeah, but you could use `State` enum members instead of bytes:

```
            if ((out.getVersion().before(Version.V_3_0_0)) && state == State.PARTIAL){
                    out.writeByte(State.SUCCESS.value());
            } else {
                    out.writeByte(state.value());
            }
```

thanks!

Reply:
Thank you so much! 
And for the in Entry(StreamInput in) method, I am guessing this would work. 
```
        private Entry(StreamInput in) throws IOException {
            snapshot = new Snapshot(in);
            ...
            if ((in.getVersion().before(Version.V_3_0_0)) && State.fromValue(in.readByte()) == State.PARTIAL){
               state = State.SUCCESS;
            } else {
                state = State.fromValue(in.readByte());
            }
```
Please let me know if you have any thoughts on this. 
Meanwhile, I will run the tests with this change and push again.

Reply:
>  And for the in Entry(StreamInput in) method, I am guessing this would work.

I think for `Entry(StreamInput in)` counterpart I misled you, apologies for that, we don't need to change it. The new code will always be able to make sense of `PARTIAL` status.